folly/net/NetOps.h

```diff
@@ -77,8 +77,10 @@ struct mmsghdr {
   unsigned int msg_len;
 };
 #else
+#ifndef FOLLY_HAVE_SENDMMSG
```

Comment:
But why is *this* the fix?

V.s. updating the `#if` line just above.

Reply:
struct mmsghdr is already defined in /usr/include/bits/socket.h.  If update #if statement, we will get 'redefinition of struct mmsghdr' compilation error.

Reply:
What did you try?

Reply:
I tried below:

#if (!__linux__) || (defined(__ANDROID__) && (__ANDROID_API__ < 21)) || (defined(HAS_NO_SENDMMSG))

Reply:
We may need to move this to a config check in `folly/CMake/FollyConfigChecks.cmake`.

Reply:
Make sense, I think we need to make below changes:

1. remove define in NetOps.h
-#else
-#define FOLLY_HAVE_SENDMMSG 1

2. add  FOLLY_HAVE_SENDMMSG  to  CMake/folly-config.h.cmake
+#cmakedefine FOLLY_HAVE_SENDMMSG 1

3. update CMake/FollyConfigChecks.cmake
+check_symbol_exists(sendmmsg sys/socket.h FOLLY_HAVE_SENDMMSG)

I can create a PR if above changes look good to you.

Reply:
Yes something like this. Can force-update this PR if desired.

Reply:
there was something wrong with my old forked repo, so I have created a new PR (1053).
https://github.com/facebook/folly/pull/1053